pkg/operator/configobservation/util.go

```diff
@@ -21,6 +21,11 @@ func ObserveField(observedConfig map[string]interface{}, val interface{}, fieldN
 	switch v := val.(type) {
 	case int64, bool:
 		err = unstructured.SetNestedField(observedConfig, v, nestedFields...)
+	case *bool:
+		if skipIfEmpty && val == nil {
+			return nil
+		}
+		err = unstructured.SetNestedField(observedConfig, *v, nestedFields...)
```

Comment:
@coreydaley being a bit paranoid. If a caller provides `skipIfEmpty=false` and `val=nil`, we risk a nil pointer dereference panic, no?

I think our best option is to throw an error:

```
if !skipEmpty && val == nil {
    return fmt.Errorf("cannot observe nil value for bool pointer field %s", fieldName)
}
```

Reply:
updated in current push